components/tabs/tab.js

```diff
@@ -3,11 +3,26 @@ import { classMap } from 'lit/directives/class-map.js';
 import { getFocusPseudoClass } from '../../helpers/focus.js';
 import { TabMixin } from './tab-mixin.js';
 
+function handleSlotChange(e) {
+	const slot = e.target;
+	if (!slot || !slot.assignedElements || slot.assignedElements().length === 0) return;
+	const elems = slot.assignedElements({ flatten: true }).filter((node) => (getComputedStyle(node).display !== 'none' || node.hasAttribute('data-hide')));
+	if (!elems) return;
+
+	for (let i = 0; i < elems.length; i++) {
+		if (i === 0) {
+			elems[i].removeAttribute('data-hide');
```

Comment:
This is effectively modifying the DOM outside of the consumer's `render` method which we shouldn't really do. If we need to allow the consumer to have hidden elements in the slot, only showing one visible, could we leverage a slotted selector using the standard `hidden` attribute instead of computing display and modifying their DOM? - if they want a hidden icon in the slot, they have to do it that way. That would have the side benefit to avoiding `getComputedStyle`, which also can have a cost.

Reply:
Yeah exactly. Jeff confirmed that he'd be okay with the css solution mentioned in the description.